packages/view-types/statistical-plots/src/ExpressionHistogramSubscriber.js

```diff
@@ -72,7 +88,10 @@ export function ExpressionHistogramSubscriber(props) {
       return Array.from(expressionData[0]).map((_, index) => {
         const value = expressionData[0][index];
         const normValue = value * 100 / 255;
-        return { value: normValue, gene: firstGeneSelected };
+        const newItem = { value: normValue, gene: firstGeneSelected };
+        dataOnSelect.push(newItem);
+        setDataOnSelect(dataOnSelect);
+        return newItem;
       });
```

Comment:
@keller-mark There is an issue with creating a new cell selection with this case, when a gene name is selected. The issue is that `data` doesn't contain the cellIds. I want to modify the code to add the cellIds, but I don't know how to get the cell ids out of the data for this case. I see that we are looping over `expressionData[0]`, but from where can I derive the cell ids? Does `expresionData[0][1]` hold the expression for cell id `1`, `expresionData[0][2]` hold the expression for cell id `2`, etc? Or something else?

Reply:
I see - I think you should be able to map over `obsIndex` to get the cell IDs as with the no-gene-selected case below.

The length of `expressionData` (once the data is loaded) should match the length of `geneSelection` and `loadedFeatureSelection`. In the most common case of a single gene being selected,  `expressionData[0]` holds a vector (of the same length as `obsIndex`) containing the expression values for all cells (corresponding to the gene name in the `firstGeneSelected` variable)